test/utils_tests.js

```diff
@@ -37,4 +37,76 @@ describe('Utils', function () {
                           'The graph has a cycle.');
         });
     });
+
+    describe('searchAttributes', function () {
```

Comment:
Would it make sense to have a negative test, that doesn't find anything? Is that a concept that makes sense here?

Reply:
Do you think its sufficient to assume the negative works given that the other tests show that it chooses the correct one out of two? (it finds that one node passes the check while the other doesn't?)